scripts/_download.js

```diff
@@ -203,3 +203,169 @@ export async function downloadAndExtractTarGz({ targetDirectory, url, sha256 })
+    for (const file of files) {
+      if (file.endsWith(".zip")) {
+        const nestedZipPath = join(tempDir, file);
+        console.log(`Extracting nested zip: ${nestedZipPath} to ${finalDir}`);
+        await unzipper.Open.file(nestedZipPath).then((d) => d.extract({ path: finalDir }));
+        console.log(`Deleting nested zip: ${nestedZipPath}`);
+        await fs.unlink(nestedZipPath);
+      }
+    }
+
+    // Cleanup temporary directory
+    console.log(`Cleaning up temporary directory: ${tempDir}`);
+    await fs.rm(tempDir, { recursive: true, force: true });
+
+    // Cleanup the original zip file
+    console.log(`Deleting original zip file: ${filePath}`);
+    await fs.unlink(filePath);
+
+    console.log(`Extraction complete: ${filePath} -> ${finalDir}`);
+  } catch (error) {
+    console.error(`Error extracting artifact ${filePath}:`, error.message);
+    throw error;
+  }
+}
+
+/**
+ * Download and process workflow artifacts.
+ *
+ * @param {string} targetDirectory - Directory to store downloaded artifacts
+ * @param {string} metaFile - Path to the metadata file
+ * @param {string} url - Base URL of the GitHub repository
+ * @param {string} workflow - Name of the workflow file
+ * @param {Array} assets - List of assets to downloa
+ */
+export async function downloadWorkflowArtifacts({
+  targetDirectory,
+  metaFile,
+  url,
+  workflow,
+  assets,
+}) {
+  const GITHUB_TOKEN = process.env.GITHUB_TOKEN;
+
+  if (!GITHUB_TOKEN) {
+    console.error("Error: GITHUB_TOKEN environment variable is not set.");
+    process.exit(1);
+  }
+
+  try {
+    console.log("Fetching first successful workflow run...");
+    const workflowRunId = await fetchFirstSuccessfulRun(url, workflow, GITHUB_TOKEN);
+
+    if (!workflowRunId) {
+      throw new Error("No successful workflow runs found.");
+    }
+
+    console.log(`Workflow Run ID: ${workflowRunId}`);
+    const artifactUrls = await fetchArtifactsForRun(url, workflowRunId, GITHUB_TOKEN);
+
+    if (!artifactUrls || artifactUrls.length === 0) {
+      throw new Error("No artifacts found for the workflow run.");
+    }
+
+    const metadata = {
+      workflowRunId,
+      collectedAt: new Date().toISOString(),
+      assets: [],
+    };
+
+    for (const asset of assets) {
+      const { name } = asset;
+      const artifact = artifactUrls.find((a) => a.name === name);
+
+      if (!artifact) {
+        console.warn(`Asset [${name}] was not found in the workflow artifacts.`);
+        continue;
+      }
+
+      try {
+        console.log(`Processing artifact: ${name}`);
+
+        const downloadedFilePath = await downloadArtifact(
+          artifact.url,
+          name,
+          targetDirectory,
+          GITHUB_TOKEN,
+        );
+
+        // Extract the artifact
+        const tempDir = join(targetDirectory, "temp");
+        const extractionDir = join(targetDirectory, name.replace(/\.zip$/, ""));
+        await extractArtifact(downloadedFilePath, tempDir, extractionDir);
+
```

Comment:
Here is where the chmod would go if not put inside the extractArtifact function itself.

Something like this would work....
```js
        for (const file of await globby(["*", "!*.zip"], { cwd: extractionDir, absolute: true })) {
          console.log(`do something with ${file}`);
        }
```

Reply:
I am not able to get the permissions changed using `chmodOwnerPlusX(path)` from `_util.js`. Looking into other options

Reply:
updated